File: frontend/src/app/services/inventario.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';

import { environment } from '../../environments/environment';
import { SucursalDisponibilidad } from '../models/catalogo';

export interface DisponibilidadFiltro {
  sucursal_id?: number | null;
  ciudad_id?: number | null;
}

@Injectable({ providedIn: 'root' })
export class InventarioService {
  private http = inject(HttpClient);

  private api = `${environment.apiUrl}/inventario`;

  disponibilidad(
    varianteId: number,
    filtro: DisponibilidadFiltro = {},
  ): Observable<SucursalDisponibilidad[]> {
    let params = new HttpParams();
    if (filtro.sucursal_id) params = params.set('sucursal_id', filtro.sucursal_id);
    if (filtro.ciudad_id) params = params.set('ciudad_id', filtro.ciudad_id);
    return this.http.get<SucursalDisponibilidad[]>(
      `${this.api}/disponibilidad`,
      { params: params.set('variante_id', varianteId) },
    );
  }
}

File: mobile/lib/services/inventario_service.dart
import '../api/api_client.dart';
import '../models/sucursal_disponibilidad.dart';

class InventarioService {
  static Future<List<SucursalDisponibilidad>> disponibilidad(
    int varianteId, {
    int? ciudadId,
  }) async {
    final query = <String, String>{
      'variante_id': '$varianteId',
      if (ciudadId != null) 'ciudad_id': '$ciudadId',
    };
    final data = await ApiClient.get('/inventario/disponibilidad', query: query);
    return (data as List)
        .map((e) => SucursalDisponibilidad.fromJson(e as Map<String, dynamic>))
        .toList();
  }
}